### PR TITLE
bos.pegtoken and unittests

### PR DESCRIPTION
**Pull request by libhasheos**
## Change Description

- Add pegtoken contracts V2 with core functionalities
- Fix the files structure issue in bos.pegtoken folder
- Add bos.pegtoken_tests in tests for unittest purpose

## Deployment Changes
- [ ] Deployment Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,9 @@
 cmake_minimum_required(VERSION 3.5)
 
-project(eosio_contracts VERSION 2.0.2)
-
+project(eosio_contracts VERSION 2.0.1)
```
Comment (vlbos) [inline CMakeLists.txt]: 2.0.1替换2.0.2不对吧

File: tests/personal.bos_tests.cpp [hunk 2]
```diff
@@ -16,133 +16,103 @@ using namespace std;
...
-
-      set_code( N(personal.bos), contracts::personal_wasm() );
-      set_abi( N(personal.bos), contracts::personal_abi().data() );
+      set_code(N(personal.bos), contracts::personal_wasm());
+      set_abi(N(personal.bos), contracts::personal_abi().data());
 
       produce_blocks();
 
-      const auto& accnt = control->db().get<account_object,by_name>( N(personal.bos) );
+      const auto &accnt = control->db().get<account_object, by_name>(N(personal.bos));
```
Comment (vlbos) [inline tests/personal.bos_tests.cpp]: 错误格式替换正确格式了

File: tests/contracts.hpp.in [hunk 3]
```diff
@@ -4,32 +4,17 @@
...
-   static std::vector<uint8_t> wrap_wasm() { return read_wasm("${CMAKE_BINARY_DIR}/../eosio.wrap/eosio.wrap.wasm"); }
-   static std::string          wrap_wast() { return read_wast("${CMAKE_BINARY_DIR}/../eosio.wrap/eosio.wrap.wast"); }
-   static std::vector<char>    wrap_abi() { return read_abi("${CMAKE_BINARY_DIR}/../eosio.wrap/eosio.wrap.abi"); }
-   static std::vector<uint8_t> bios_wasm() { return read_wasm("${CMAKE_BINARY_DIR}/../eosio.bios/eosio.bios.wasm"); }
-   static std::string          bios_wast() { return read_wast("${CMAKE_BINARY_DIR}/../eosio.bios/eosio.bios.wast"); }
-   static std::vector<char>    bios_abi() { return read_abi("${CMAKE_BINARY_DIR}/../eosio.bios/eosio.bios.abi"); }
-   static std::vector<uint8_t> personal_wasm() { return read_wasm("${CMAKE_BINARY_DIR}/../personal.bos/personal.bos.wasm"); }
-   static std::string          personal_wast() { return read_wast("${CMAKE_BINARY_DIR}/../personal.bos/personal.bos.wast"); }
-   static std::vector<char>    personal_abi() { return read_abi("${CMAKE_BINARY_DIR}/../personal.bos/personal.bos.abi"); }
```
Comment (vlbos) [inline tests/contracts.hpp.in]: 删除了

File: tests/personal.bos_tests.cpp [hunk 4]
```diff
@@ -16,133 +16,103 @@ using namespace std;
...
+try
+{
    //only owner can call setpersonal
    BOOST_REQUIRE_EQUAL(error("missing authority of alice"),
-                        push_action( N(bob), N(setpersonal), mvo()
-                                          ("account", "alice")
-                                          ("key", "favorfood")
-                                          ("value", "pizza"))
-   );
+                       push_action(N(bob), N(setpersonal), mvo()("account", "alice")("key", "favorfood")("value", "pizza")));
```
Comment (vlbos) [inline tests/personal.bos_tests.cpp]: format 成错误格式了

Comment (libhasheos) [reply tests/personal.bos_tests.cpp]: OK

File: tests/contracts.hpp.in (hunk 3 above)
Comment (libhasheos) [reply tests/contracts.hpp.in]: 已修复

File: tests/personal.bos_tests.cpp (hunk 2 above)
Comment (libhasheos) [reply tests/personal.bos_tests.cpp]: 已修复

File: CMakeLists.txt (hunk 1 above)
Comment (libhasheos) [reply CMakeLists.txt]: 已修复

**Comment by Thaipanda** [comment]
@libhasheos  please recommit latest code

**Pull request closed without merging**